armstrongsouljah/StackOverflow-Clone
# #167557532 Implement User Signup

#### What does this Pr do?
- Implements custom user authentication using jwt.

#### Description of the task?
- Create a custom model
- Create an authentication backend
- Write endpoints for login and register

#### How can this be manually tested?
- Fetch this branch locally
- Check out the branch
- Install the dependencies
- Start the development server
- Visit endpoint `localhost:8000/auth/user/` to sign up
_Sample request body
{
 "user": {
      "email": "mail@server.com",
      "password": "password"
     }
}

- Visit endpoint `localhost:8000/auth/login/` to login
_Sample request body
{
 "user": {
      "email": "mail@server.com",
      "password": "password"
     }
}

#### What are the related stories?
[#167557532](https://www.pivotaltracker.com/story/show/167557532)
